### ISSUE TITLE
Inspector.print Recurses Forever

### ISSUE THREAD

**Issue by gili-labs** [nalexn/ViewInspector#105]
Ever have a problem with `Inspector.print(_:)` recursing forever?

I have the following view modifier:

```Swift
@available(iOS 13.0, macOS 10.15, tvOS 13.0, *)
private struct InspectableTestModifier: ViewModifier, Inspectable {
    
    @State private(set) var flag: Bool
    
    let publisher = PassthroughSubject<Bool, Never>()
    let inspection = InspectionForViewModifier<Self>()
    var didAppear: ((Self.Body) -> Void)?
    
    func body(content: Self.Content) -> some View {
        HStack {
            content
            Button(
                action: { self.flag.toggle() },
                label: { Text(flag ? "true" : "false") }
            )
        }
        .onReceive(publisher) { self.flag = $0 }
        .onReceive(inspection.notice) { self.inspection.visit(self.body(content: content), $0) }
        .onAppear { self.didAppear?(self.body(content: content)) }
    }
}
```

And the following test:

```Swift
    func testViewModifierOnFunction() throws {
        var sut = InspectableTestModifier(flag: false)
        let exp = sut.on(\.didAppear) { view in
            print(Inspector.print(view))
        }
        let view = EmptyView().modifier(sut)
        ViewHosting.host(view: view)
        wait(for: [exp], timeout: 0.1)
    }

```

`Inspector.print(_:)` recurses until the stack runs out.

I've been trying to debug this, and I think at some point the mirror has a child that is a reference to an ancestor. After spending the day on this, I thought I would post this issue and see if anyone has seen this before.

**Comment by nalexn** [comment]
As discussed in #110 and #108, we are in need for a non-recusrive approach for `ViewModifier` inspection

**Comment by nalexn** [comment]
The v0.8.0 release does not eliminate the recursion, which is necessary for the other parts to work properly. I'll leave this open for now. The snippets updated for 0.8.0 are:

```swift
@available(iOS 13.0, macOS 10.15, tvOS 13.0, *)
private struct InspectableTestModifier: ViewModifier, Inspectable {
    
    @State private(set) var flag: Bool
    
    let publisher = PassthroughSubject<Bool, Never>()
    let inspection = Inspection<Self>()
    var didAppear: ((Self) -> Void)?
    
    func body(content: Self.Content) -> some View {
        HStack {
            content
            Button(
                action: { self.flag.toggle() },
                label: { Text(flag ? "true" : "false") }
            )
        }
        .onReceive(publisher) { self.flag = $0 }
        .onReceive(inspection.notice) { self.inspection.visit(self, $0) }
        .onAppear { self.didAppear?(self) }
    }
}
```

```swift
func testViewModifierOnFunction() throws {
    var sut = InspectableTestModifier(flag: false)
    let exp = sut.on(\.didAppear) { view in
        print(Inspector.print(view))
    }
    let view = EmptyView().modifier(sut)
    ViewHosting.host(view: view)
    wait(for: [exp], timeout: 0.1)
}
```

**Comment by nalexn** [comment]
This has been fixed in [0.9.0](https://github.com/nalexn/ViewInspector/releases/tag/0.9.0)
